### ISSUE TITLE
`RRuleSet`s don't handle `DTSTART` correctly

### ISSUE THREAD

**Issue by topherauyeung** [jakubroztocil/rrule#413]
### Reporting an issue
When using `rrulestr` to parse a set of rules, creating an `RRuleSet` and then enumerating its instances is not working as expected.

```
rrule.rrulestr(["RRULE:FREQ=DAILY;COUNT=2"].join("\n"), { dtstart: new Date(Date.UTC(2020, 6, 4, 0, 0)) }).toString()
```
produces
```
'DTSTART:20200704T000000Z\nRRULE:FREQ=DAILY;COUNT=2'
```

```
rrule.rrulestr(["RRULE:FREQ=DAILY;COUNT=2",  "EXDATE:20600130T124000Z"].join("\n"), { dtstart: new Date(Date.UTC(2020, 6, 4, 0, 0)) 
```
produces
```
'RRULE:FREQ=DAILY;COUNT=2\nEXDATE:20600130T124000Z'
```
and when inspecting the resultant `RRuleSet`, the `dtstart` parameter is not set correctly.

Notice that the `DTSTART` is missing entirely, because in the second example, `rrulestr` is producing an `RRuleSet` instead of an `RRule`. This bug also occurs if you pass `forceset: true` into the `rrulestr` options.

Version of rrule: 2.6.4
Operating system: MacOS High Sierra
Local timezone: PDT

**Comment by Noitidart** [comment]
Confirming, I have the same problem. Took a long to figure this one out.

**Comment by Noitidart** [comment]
I wrote a workaround until then, it requires moment.

```
function getSetFromTimeRule(timeRule, options = {}) {
  let dtstartStr = '';
  if (options.dtstartMoment) {
    dtstartStr =
      'DTSTART:' + startWeekMoment.format('YYYYMMDD[T]HHmmss[Z]') + '\n';
  }

  const set = rrulestr(dtstart + timeRule);

  return set;
}
```

**Comment by PanMan** [comment]
Thanks! I had the same issue: setting `dtstart` via the `rrulestr` `options` didn't work: I got dates before the DTSTART in the result.
The fix above by @Noitidart does work, but seems a workaround for broken functionality.

**Comment by Noitidart** [comment]
Agreed, I really dont like my work around. I should remove moment from it too, its too much for others that use it and dont have moment installed.

**Comment by ephys** [comment]
This sounds like basically the same issue as https://github.com/jakubroztocil/rrule/issues/332#issuecomment-811212174
I've posted a reimplementation of the parser there

**Comment by game-of-morgan** [comment]
Hey folks,

I had this issue when using the rrulestr function with an RRULE that contained an EXDATE. The solution for me was to prefer using `parsedRule.dtstart()` and `parsedRule.tzid()`. I now fallback to what I was previously using (broken) which is `parsedRule.options.dtstart` and `parsedRule.options.tzid`.

It seems that the callable functions (which I hope are always present) work more reliably than accessing the options.

**Comment by NOPR9D** [comment]
So ? No news ?

![giphy (11)](https://user-images.githubusercontent.com/75882457/227580595-897a86c0-9dd5-40a2-8947-c6833dc5656b.gif)
